### PR TITLE
Atf fuzz

### PR DESCRIPTION
**Pull request by alexeilebedev**

### CODE REVIEW

File: cpp/atf/atf_fuzz.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    }ind_end;
+    prlog("atf_fuzz.skip_inputs.begin"
+          <<Keyval("inputfile",_db.cmdline.inputfile)
+          <<Keyval("reprofile",_db.cmdline.reprofile)
+          <<Keyval("nline",nline));
+    DeleteFile(_db.cmdline.reprofile);
+    int ncrash=0;
+    prlog("atf_fuzz: preparing sandbox");
+    strptr sandbox="temp/atf_fuzz.sandbox/";
+    SysCmd(tempstr()<<"rsync -ac --exclude .git --exclude temp ./ "<<sandbox, FailokQ(false));
```
Comment (ali5h) [inline cpp/atf/atf_fuzz.cpp:35]: add `--filter=':- .gitignore'`

Comment (alexeilebedev) [reply cpp/atf/atf_fuzz.cpp:35]: I need the build directory. I can still try to base the exclude list on .gitignore

Comment (ali5h) [reply cpp/atf/atf_fuzz.cpp:35]: add it via `--include=build`

**Review by ali5h — APPROVED** [review]

File: cpp/atf/atf_fuzz.cpp [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+          <<Keyval("inputfile",_db.cmdline.inputfile)
+          <<Keyval("reprofile",_db.cmdline.reprofile)
+          <<Keyval("nline",nline));
+    DeleteFile(_db.cmdline.reprofile);
+    int ncrash=0;
+    prlog("atf_fuzz: preparing sandbox");
+    strptr sandbox="temp/atf_fuzz.sandbox/";
+    command::bash_proc init;
+    init.cmd.c << "(grep -v build .gitignore; echo .git) > temp/atf_fuzz.filter" << eol;
+    init.cmd.c << "rsync -ac --filter=':- temp/atf_fuzz.filter' ./ "<<sandbox;
```
Comment (ali5h) [inline cpp/atf/atf_fuzz.cpp:37]: i don't think `:` is needed anymore, it was to merge all `.gitignore` files in the tree. now there is just one filter file

File: cpp/atf/atf_fuzz.cpp [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    vrfy(bash_Exec(init)==0,tempstr()<<"failed: "<<init.cmd.c);
+    int iter=0;
+    ind_beg(Line_curs,line,inputs) {
+        if (algo::double_WeakRandom(1) <= _db.cmdline.testprob) {
```
Comment (ali5h) [inline cpp/atf/atf_fuzz.cpp:41]: nice

**Pull request merged**